File: e_rents_desktop/lib/models/property_tenant_summary.dart
class PropertyTenantSummary {
  final int tenantId;
  final int userId;
  final String? fullName;
  final String? email;
  final DateTime? leaseStartDate;
  final DateTime? leaseEndDate;
  final String tenantStatus; // keep as string for simplicity

  const PropertyTenantSummary({
    required this.tenantId,
    required this.userId,
    this.fullName,
    this.email,
    this.leaseStartDate,
    this.leaseEndDate,
    required this.tenantStatus,
  });

  factory PropertyTenantSummary.fromJson(Map<String, dynamic> json) {
    DateTime? _parse(dynamic v) {
      if (v == null) return null;
      final s = v.toString();
      if (s.isEmpty) return null;
      return DateTime.tryParse(s);
    }
    return PropertyTenantSummary(
      tenantId: (json['tenantId'] as num).toInt(),
      userId: (json['userId'] as num).toInt(),
      fullName: json['fullName'] as String?,
      email: json['email'] as String?,
      leaseStartDate: _parse(json['leaseStartDate']),
      leaseEndDate: _parse(json['leaseEndDate']),
      tenantStatus: (json['tenantStatus']?.toString() ?? ''),
    );
  }
}

File: eRents.Features/PropertyManagement/Models/PropertyTenantSummary.cs
using System;
using eRents.Domain.Models.Enums;

namespace eRents.Features.PropertyManagement.Models;

public sealed class PropertyTenantSummary
{
    public int TenantId { get; set; }
    public int UserId { get; set; }
    public string? FullName { get; set; }
    public string? Email { get; set; }
    public DateOnly? LeaseStartDate { get; set; }
    public DateOnly? LeaseEndDate { get; set; }
    public TenantStatusEnum TenantStatus { get; set; }
}
